cfme/tests/intelligence/reports/test_metering_report.py

```diff
@@ -33,238 +25,26 @@
-        'ems_metrics_coordinator', 'ems_metrics_collector')
-    # Perfrom rollup of C&U data.
-    ret = appliance.ssh_client.run_rails_command(
-        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
-        vm.perf_rollup_range(2.hour.ago.utc, Time.now.utc,'realtime')\"".
-        format(provider.id, repr(vm_name)))
-    assert ret.success, "Failed to rollup VM C&U data:".format(ret.out)
-
-    wait_for(verify_records_rollups_table, [appliance, provider, vm_name], timeout=600,
-        fail_condition=False, message='Waiting for hourly rollups')
-
-    # Since we are collecting C&U data for > 1 hour, there will be multiple hourly records per VM
-    # in the metric_rollups DB table.The values from these hourly records are summed up.
-
-    with appliance.db.client.transaction:
-        result = (
-            appliance.db.client.session.query(rollups.id)
-            .join(ems, rollups.parent_ems_id == ems.id)
-            .filter(rollups.capture_interval_name == 'hourly',
-                    rollups.resource_name == vm_name,
-                    ems.name == provider.name, rollups.timestamp >= date.today())
-        )
-
-    for record in appliance.db.client.session.query(rollups).filter(
-            rollups.id.in_(result.subquery())):
-        cpu_used_in_mhz = cpu_used_in_mhz + record.cpu_usagemhz_rate_average
-        memory_used_in_mb = memory_used_in_mb + record.derived_memory_used
-        network_io = network_io + record.net_usage_rate_average
-        disk_io = disk_io + record.disk_usage_rate_average
-        storage_used = storage_used + record.derived_vm_used_disk_storage
-
-    # Convert storage used in Bytes to GB
-    storage_used = storage_used * math.pow(2, -30)
-
-    return {"cpu_used": cpu_used_in_mhz,
-            "memory_used": memory_used_in_mb,
-            "network_io": network_io,
-            "disk_io_used": disk_io,
-            "storage_used": storage_used}
-
-
-@pytest.fixture(scope="module")
-def metering_report(appliance, vm_ownership, provider):
-    # Create a Metering report based on VM owner; Queue the report.
-    owner = vm_ownership
-    data = {
-        'menu_name': 'cb_' + provider.name,
-        'title': 'cb_' + provider.name,
-        'base_report_on': 'Metering for VMs',
-        'report_fields': ['Owner', 'Memory Used',
-        'CPU Used', 'Disk I/O Used',
-        'Network I/O Used', 'Storage Used',
-        'Existence Hours Metric', 'Metering Used Metric'],
-        'filter': {
-            'filter_show_costs': 'Owner',
-            'filter_owner': owner,
-            'interval_end': 'Today (partial)'
-        }
-    }
-    report = appliance.collections.reports.create(is_candu=True, **data)
-
-    logger.info('Queuing Metering report for {} provider'.format(provider.name))
-    report.queue(wait_for_finish=True)
-
-    yield list(report.saved_reports.all()[0].data.rows)
-    report.delete()
+def get_report_value(report, report_key):
+    report_value = None
```

Comment:
docblock here please, for the intended use of this. Why break on finding the first group with the report_key? What's the handling of the KeyError on `groups[report_key]` if its not there?